### PR TITLE
Switch to Scala 3 by default

### PR DESCRIPTION
**Pull request by romanowski**

### CODE REVIEW

File: build.sc [hunk 1]
```diff
@@ -653,7 +655,7 @@ trait Cli3 extends Cli {
       mainArgs = Seq(cache.toNIO.toString, classpath),
       workingDir = os.pwd
     )
-    val cp = res.out.text
+    val cp = res.out.lines.head
```
Comment (lwronski) [inline build.sc]: Are you sure lines will contain at least one element? maybe it would be better to throw a descriptive error

**Review by lwronski — COMMENTED** [review]
I think we should also update logic in method `scalaCliScalaVersion` which resolves the scala version used to build scala-cli. 

IMO we can just add new `else if` condition, like this:
```
if (cliVersion == "nightly") Properties.versionNumberString
    else if (Version(cliVersion) <= Version("0.1.2")) Constants.defaultScala212Version
    else if (Version(cliVersion) <= Version("0.1.4")) Constants.defaultScala213Version
    else Properties.versionNumberString
```

File: build.sc (hunk 1 above)
Comment (romanowski) [reply build.sc]: I've added a comment to address your point

File: build.sc [hunk 2]
```diff
@@ -134,7 +136,7 @@ object packager extends ScalaModule with Bloop.Module {
 }
 
 object `generate-reference-doc` extends SbtModule with ScalaCliScalafixModule {
-  def scalaVersion = Scala.defaultInternal
+  def scalaVersion = Scala.scala3
```
Comment (alexarchambault) [inline build.sc]: Why not change the value of `defaultInternal` to `scala3`? (and change `Scala.defaultInternal` to `Scala.scala2` when needed)

File: build.sc (hunk 1 above)
Comment (alexarchambault) [reply build.sc]: Wouldn't calling `.trim` on `res.out.text` just work?

File: build.sc (hunk 2 above)
Comment (romanowski) [reply build.sc]: I wanted to do it as next step - I want to first migrate to Scala 3 and then when all is working clean up the build

**Review by alexarchambault — APPROVED** [review]

**Pull request merged**